examples/tracing-http-propagator/src/server.rs

```diff
@@ -44,7 +44,9 @@ async fn handle_echo(req: Request<Body>) -> Result<Response<Body>, Infallible> {
     Ok(res)
 }
 
-async fn router(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+async fn router(
+    req: Request<Incoming>,
+) -> Result<Response<Either<Full<Bytes>, Incoming>>, Infallible> {
```

Comment:
Fair, I personally don't find the type signature *too* hard to read though, and we save an allocation (and pointer indirection through dynamic dispatch) without too much of a downside.

I'm open to changing it though 🤷‍♀️

Reply:
It's not really about the type, more the usage. You have two different methods where each returns a different type and then you have to map the response to wrap it in either left or right and then you even have to do the same thing on the top level to return 404.

Personally, I think that this is just distracting from the purpose of the example which is to show how to extract context from incoming requests. I'd even consider just collecting the bytes in the echo handler and just returning `Response<Full<Bytes>>`. Because performance is not the point at all here. Heck, I'd even vote to remove one of the handlers because it doesn't help the example at all.

Again, feel free to keep what you have and ignore this, it's just opinions.

Reply:
Okay, I really overlooked that this was just an example. I agree, in the case of an example, performance (especially such nit-picky optimizations) aren't really necessary.  

I'll probably just replace it with a boxed body

Reply:
Well, just replaced it and I'm not entirely sure that that's an improvement.

All in all, the example became *longer* and it now has to showcase the idea that `|err: Infallible| match err {}` resolves to the bottom (never) type, and therefore can be used to coerce it into an `hyper::Error`.